### PR TITLE
Adding new R2D2 V2 code to EWOK tasks.

### LINKED ISSUE #854: MySQL version of EWOK R2D2 calls
## Description

This issue addresses the need to replace the old R2D2 store and fetch calls with the new V2 MySQL-based API.

### PR DESCRIPTION
**Pull request by ericlingerfelt**
## Description

This pull request replaces the old R2D2 store and fetch calls with the new V2 MySQL-based API. 

## Description

This pull request replaces the old R2D2 store and fetch calls with the new V2 MySQL-based API. 

### Associated pull requests

To be merged with 
- https://github.com/JCSDA-internal/r2d2-data/pull/26
- https://github.com/JCSDA-internal/ewok/pull/470
- https://github.com/JCSDA-internal/r2d2/pull/260
- https://github.com/JCSDA-internal/fv3-jedi/pull/705

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #854

### CODE REVIEW

File: ewok/tasks/getBackgroundMOM6.py [hunk 1]
```diff
@@ -15,4 +15,8 @@ def setup(self, config, fc):
         super().setup(config, fc)
 
         # Use MOM6 specific script
-        self.command = os.path.join(config['model_path'], "tasks/runGetForecast.py")
\ No newline at end of file
+        self.command = os.path.join(config['model_path'], "tasks/runGetForecast.py")
+
+        self.exec_cmd = ''   # Run on login node for S3 and R2D2 Database access
+        self.include_header = ''
+        self.login_node_limit = 'True'
```
Comment (cmgas) [inline ewok/tasks/getBackgroundMOM6.py:22]: 🎉

**Review by cmgas — COMMENTED** [review]
Code looks good, I will test tomorrow

**Comment by cmgas** [comment]
```
Traceback (most recent call last):
  File "/Users/cgas/Projects/JEDI/tmp/ecflow/bf5fec/an/runGetForecast.py", line 40, in <module>
    R2D2Data.fetch(
  File "/Users/cgas/Projects/JEDI/jedi-bundle/r2d2/src/r2d2/r2d2_data.py", line 175, in fetch
    r2d2_data.validate_kwargs(kwargs)
  File "/Users/cgas/Projects/JEDI/jedi-bundle/r2d2/src/r2d2/r2d2_item.py", line 135, in validate_kwargs
    self.process_kwargs(kwargs)
  File "/Users/cgas/Projects/JEDI/jedi-bundle/r2d2/src/r2d2/r2d2_item.py", line 188, in process_kwargs
    R2D2Index.process_index_item_kwarg(kwargs, item)
  File "/Users/cgas/Projects/JEDI/jedi-bundle/r2d2/src/r2d2/r2d2_index.py", line 54, in process_index_item_kwarg
    raise err.RegistrationNotFound(item, kwargs[item])
r2d2.error.RegistrationNotFound.RegistrationNotFound: 
dummy is not registered in experiment yet!
You must manually register this Name using R2D2Index.register() method.
```

The code looks good but the experiment doesn't seem to exist

**Comment by ericlingerfelt** [comment]
> ```
> Traceback (most recent call last):
>   File "/Users/cgas/Projects/JEDI/tmp/ecflow/bf5fec/an/runGetForecast.py", line 40, in <module>
>     R2D2Data.fetch(
>   File "/Users/cgas/Projects/JEDI/jedi-bundle/r2d2/src/r2d2/r2d2_data.py", line 175, in fetch
>     r2d2_data.validate_kwargs(kwargs)
>   File "/Users/cgas/Projects/JEDI/jedi-bundle/r2d2/src/r2d2/r2d2_item.py", line 135, in validate_kwargs
>     self.process_kwargs(kwargs)
>   File "/Users/cgas/Projects/JEDI/jedi-bundle/r2d2/src/r2d2/r2d2_item.py", line 188, in process_kwargs
>     R2D2Index.process_index_item_kwarg(kwargs, item)
>   File "/Users/cgas/Projects/JEDI/jedi-bundle/r2d2/src/r2d2/r2d2_index.py", line 54, in process_index_item_kwarg
>     raise err.RegistrationNotFound(item, kwargs[item])
> r2d2.error.RegistrationNotFound.RegistrationNotFound: 
> dummy is not registered in experiment yet!
> You must manually register this Name using R2D2Index.register() method.
> ```
> 
> The code looks good but the experiment doesn't seem to exist

Are you running this locally using r2d2-data? If so, then there are no entries for SOCA there. You have to use EWOK etc on an HPC to get access to these entries from the main database on AWS. You just need to have R2D2_HOST set. Also, I have to re-ingest the mom6 backgrounds.

**Comment by cmgas** [comment]
I was trying locally yes

**Comment by ericlingerfelt** [comment]
And you will need to have the AWS `credentials` and `config` files updated on the HPC to the ones in the tutorial. In the end, the only difference between this code and the other code is that there is a `domain='global'` key/value for the `forecast` and `analysis` files. The new R2D2 uses your AWS credentials to download the R2D2 database credentials from an S3 bucket.

**Review by cmgas — APPROVED** [review]

**Review by ashley314 — APPROVED** [review]
Reviewed code and syntax for accuracy. Looks good, but I was not able to test.

**Comment by travissluka** [comment]
> Are you running this locally using r2d2-data? If so, then there are no entries for SOCA there. You have to use EWOK etc on an HPC to get access to these entries from the main database on AWS. You just need to have R2D2_HOST set. Also, I have to re-ingest the mom6 backgrounds.

@ericlingerfelt are the background re-ingested yet? I'm trying to run on orion now and the `mom6-` experiments are returning

```
r2d2.error.RecordNotFound.RecordNotFound: 
Record does not exist in MySQL Database.
Table Name: forecast
Query:
SELECT `forecast_index` FROM `forecast` WHERE `step` = P1D AND `resolution` = 025deg AND `date` = 2021-07-31 12:00:00 AND `domain` = global AND `file_type` = MOM.res AND `tile` = -9999 AND `member` = -9999 AND `experiment_index` = 357 AND `file_extension_index` = 162 AND `model_index` = 6
```

**Comment by ericlingerfelt** [comment]
Not yet. Actually I just need to update the database records and not the files themselves. That is, once I get that done, we will not have to re-sync them across the platforms. I'll try my best to get that done next in my long list tasks.

**Pull request merged**